### PR TITLE
Create certificate from .cer and .key files for Rest API

### PR DESCRIPTION
**Pull request by arvindgh**

### CODE REVIEW

File: Ginger/GingerCoreNET/ActionsLib/WebAPI/ActWebAPIBase.cs [hunk 1]
```diff
@@ -87,6 +87,7 @@ public new static partial class Fields
 
             public static string UseLegacyJSONParsing = "UseLegacyJSONParsing";
 
+            public static string KeyFilePath = "KeyFilePath";
```
Comment (ambarkhan) [inline Ginger/GingerCoreNET/ActionsLib/WebAPI/ActWebAPIBase.cs:90]: Codacy found an issue: [Change the visibility of 'KeyFilePath' or make it 'const' or 'readonly'.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6345816)

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6345816&bid=20608809)

File: Ginger/Ginger/Actions/ActionEditPages/WebServices/ActWebAPIEditPage.xaml [hunk 2]
```diff
@@ -179,6 +179,10 @@
                                 <Actions:UCValueExpression x:Name="CertificatePath" Margin="0,0,0,0"/>
                                 <CheckBox x:Name="DoNotCertificateImportFile" Content="Import Certificate file to Solution folder" ToolTip="If checked the selected certificate file will be imported to ~Documents\Certificates\" HorizontalAlignment="Left" Margin="0,1,0,0" Style="{StaticResource @InputCheckBoxStyle}"/>
                             </StackPanel>
+                            <StackPanel Orientation="Vertical">
+                                <Actions:UCValueExpression x:Name="KeyFilePathTextBox" Margin="0,0,0,0"/>
```
Comment (JinendraGhodke) [inline Ginger/Ginger/Actions/ActionEditPages/WebServices/ActWebAPIEditPage.xaml:183]: we use x as prefix to all WFP control names e.g. xKeyFilePathTextBox

File: Ginger/Ginger/Actions/ActionEditPages/WebServices/ActWebAPIEditPage.xaml [hunk 3]
```diff
@@ -179,6 +179,10 @@
...
+                                <CheckBox x:Name="DoNotImportKeyFile" Content="Import key file to Solution folder" ToolTip="If checked the selected certificate file will be imported to ~Documents\Certificates\" HorizontalAlignment="Left" Margin="0,1,0,0" Style="{StaticResource @InputCheckBoxStyle}"/>
```
Comment (JinendraGhodke) [inline Ginger/Ginger/Actions/ActionEditPages/WebServices/ActWebAPIEditPage.xaml:184]: Check box name is confusing. Content says import and check box says do not import. 
Also follow naming convension

File: Ginger/Ginger/packages.config [hunk 4]
```diff
@@ -88,14 +88,14 @@
...
-  <package id="System.Text.RegularExpressions" version="4.3.1" targetFramework="net461" />
+  <package id="System.Text.RegularExpressions" version="4.3.1" targetFramework="net461" requireReinstallation="true" />
   <package id="System.Threading" version="4.3.0" targetFramework="net461" />
   <package id="System.Threading.Tasks" version="4.3.0" targetFramework="net461" />
   <package id="System.Threading.Tasks.Extensions" version="4.5.4" targetFramework="net461" />
   <package id="System.Threading.Tasks.Parallel" version="4.3.0" targetFramework="net461" />
   <package id="System.Threading.Thread" version="4.3.0" targetFramework="net461" />
   <package id="System.Threading.Timer" version="4.3.0" targetFramework="net461" />
-  <package id="System.ValueTuple" version="4.5.0" targetFramework="net461" />
+  <package id="System.ValueTuple" version="4.5.0" targetFramework="net461" requireReinstallation="true" />
```
Comment (JinendraGhodke) [inline Ginger/Ginger/packages.config:98]: what are these changes with requirereinstallation attribute ?

File: Ginger/GingerCore/Drivers/WebServicesDriverLib/HttpWebClientUtils.cs [hunk 5]
```diff
@@ -256,10 +272,22 @@ private bool SetCertificates()
                     mAct.ExInfo = "Certificate path is missing";
                     return false;
                 }
+
             }
             return true;
         }
 
+        private static byte[] UnPem(string pem)
+        {
```
Comment (JinendraGhodke) [inline Ginger/GingerCore/Drivers/WebServicesDriverLib/HttpWebClientUtils.cs:281]: What is this function used for ?

File: Ginger/GingerCore/Properties/Resources.Designer.cs [hunk 6]
```diff
@@ -563,26 +569,19 @@ internal class Resources {
...
-        ///GingerPayLoadObj = null;
-        ///
-        ///// strict mode = must define variables which is good programming :)
-        /////&apos;use strict&apos;;
-        ///
-        ///
-        ///function GingerPayLoad(Name) {
-        ///    const LastByteMarker = 255;
-        ///    const cNULLStringLen = -1;
+        ///   Looks up a localized string similar to /*
```
Comment (JinendraGhodke) [inline Ginger/GingerCore/Properties/Resources.Designer.cs:572]: Is this change relevant to this PR?

File: Ginger/GingerCore/Properties/Settings.Designer.cs [hunk 7]
```diff
@@ -12,7 +12,7 @@ namespace GingerCore.Properties {
     
     
     [global::System.Runtime.CompilerServices.CompilerGeneratedAttribute()]
-    [global::System.CodeDom.Compiler.GeneratedCodeAttribute("Microsoft.VisualStudio.Editors.SettingsDesigner.SettingsSingleFileGenerator", "12.0.0.0")]
+    [global::System.CodeDom.Compiler.GeneratedCodeAttribute("Microsoft.VisualStudio.Editors.SettingsDesigner.SettingsSingleFileGenerator", "16.6.0.0")]
```
Comment (JinendraGhodke) [inline Ginger/GingerCore/Properties/Settings.Designer.cs:15]: Is it related to PR ?

**Review by JinendraGhodke — CHANGES_REQUESTED** [review]

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6345816&bid=20608809)

File: Ginger/Ginger/Actions/ActionEditPages/WebServices/ActWebAPIEditPage.xaml.cs [hunk 8]
```diff
@@ -343,48 +349,98 @@ private void SecCollapsed(object sender, RoutedEventArgs e)
...
                 if (ImportFileFlag)
                 {
-                    //TODO import Certificate File to solution folder
-                    string targetPath = System.IO.Path.Combine(SolutionFolder, @"Documents\WebServices\Certificates");
-                    if (!System.IO.Directory.Exists(targetPath))
-                    {
-                        System.IO.Directory.CreateDirectory(targetPath);
-                    }
+                    ////TODO import Certificate File to solution folder
+                    //string targetPath = System.IO.Path.Combine(SolutionFolder, @"Documents\WebServices\Certificates");
```
Comment (ambarkhan) [inline Ginger/Ginger/Actions/ActionEditPages/WebServices/ActWebAPIEditPage.xaml.cs:368]: Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6345816)

File: Ginger/GingerCoreNET/ActionsLib/WebAPI/ActWebAPIBase.cs [hunk 9]
```diff
@@ -87,6 +87,9 @@ public new static partial class Fields
...
+            public static string ImportKeyFile = "ImportKeyFile";
```
Comment (ambarkhan) [inline Ginger/GingerCoreNET/ActionsLib/WebAPI/ActWebAPIBase.cs:91]: Codacy found an issue: [Change the visibility of 'ImportKeyFile' or make it 'const' or 'readonly'.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6345816)

File: Ginger/GingerCoreNET/ActionsLib/WebAPI/ActWebAPIBase.cs [hunk 10]
```diff
@@ -87,6 +87,9 @@ public new static partial class Fields
 
             public static string UseLegacyJSONParsing = "UseLegacyJSONParsing";
 
+            public static string KeyFilePath = "KeyFilePath";
```
Comment (ambarkhan) [inline Ginger/GingerCoreNET/ActionsLib/WebAPI/ActWebAPIBase.cs:90]: Codacy found an issue: [Change the visibility of 'KeyFilePath' or make it 'const' or 'readonly'.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6345816)

**Pull request closed without merging**